client/gutenberg/extensions/videopress/edit.js

```diff
@@ -0,0 +1,77 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import debugFactory from 'debug';
+import getVideoPressUrl from './get-videopress-url';
+
+const debug = debugFactory( 'extensions:gutenberg:videopress' );
+
+const withVideoPressEdit = createHigherOrderComponent( CoreVideoEdit => {
+	return class VideoPressEdit extends Component {
+		setAttributes = attributes => {
+			const { id, src } = attributes;
+			const { id: currentId } = this.props.attributes;
+			let shouldSetSrcFromVideoPress = false;
```

Comment:
we can remove this mutation and simplify the control-flow through the function if we use an early-abort style. if there were fewer lines in-between this flag and we weren't setting it from within a conditional structure I don't think I'd notice it as much, but I do love to void these in order to reduce the amount of compiling/interpreting I have to do in my head when I read code

```js
if ( ! id || id === currentId ) {
	return this.props.setAttributes( attributes );
}

this.setSrcFromVideoPress( id ).catch( error => {
	…

	this.props.setAttributes( { ...attributes, src } );
} );
```